src/main/scala/org/broadinstitute/dsde/firecloud/core/AgoraPermissionHandler.scala

```diff
@@ -0,0 +1,307 @@
+import akka.event.Logging
+import org.broadinstitute.dsde.firecloud.model.MethodRepository.{AgoraPermission, FireCloudPermission}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.service.FireCloudRequestBuilding
+import org.broadinstitute.dsde.firecloud.service.PerRequest.RequestComplete
+import org.broadinstitute.dsde.firecloud.HttpClient
+import spray.client.pipelining._
+import spray.http.StatusCodes._
+import spray.http.{Uri, HttpResponse, StatusCodes}
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+import spray.routing.RequestContext
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object AgoraPermissionHandler {
+  case class Get(url: String)
+  case class Post(url: String, agoraPermissions: List[AgoraPermission])
+  def props(requestContext: RequestContext): Props = Props(new GetEntitiesWithTypeActor(requestContext))
+
+  def testFromFCToAgora():Boolean = {
+    val aReader=FireCloudPermission(
+      user=Some("a_reader"),
+      role=Some("READER"))
+    val agoraReader=aReader.toAgoraPermission
+    println("For reader "+aReader+" the agora translation is "+agoraReader)
+
+    val anOwner=FireCloudPermission(
+      user=Some("an_owner"),
+      role=Some("OWNER"))
+    val agoraOwner=anOwner.toAgoraPermission
+    println("For owner "+anOwner+" the agora translation is "+agoraOwner)
+
+    val aNoAccess=FireCloudPermission(
+      user=Some("an_owner"),
+      role=Some("NO ACCESS"))
+    val agoraNone=aNoAccess.toAgoraPermission
+    println("For owner "+aNoAccess+" the agora translation is "+agoraNone)
+
+    return true
+  }
+
+
+  def generateRandomAgoraRoles():List[String] =
+  {
+    //each of the 5 individual roles
+    //has a 1/2 probability of showing up
+    val randomNum=scala.util.Random.nextInt(2)
+    //println("A randomNum is "+randomNum)
+    var aList=List[String]()
+    if(randomNum ==0) {
+      aList=(aList :+ "Read")
+    }
+    val randomNum2=scala.util.Random.nextInt(2)
+    if(randomNum2 ==0 )
+    {
+      aList=(aList :+ "Write")
+    }
+    val randomNum3=scala.util.Random.nextInt(2)
+    if(randomNum3  ==0)
+    {
+      aList=(aList :+ "Create")
+    }
+    val randomNum4=scala.util.Random.nextInt(2)
+    if(randomNum4 ==0)
+    {
+      aList=(aList :+ "Redact")
+    }
+    val randomNum5=scala.util.Random.nextInt(2)
+    if(randomNum5 ==0)
+    {
+      aList=(aList :+ "Manage")
+    }
+    return aList
+  }
+
+
+  def testFromRandomAgoraToFC():Boolean ={
+    for( a <- 1 to 256){
+      val randomRoles=generateRandomAgoraRoles()
+      val randomAgoraPerm=AgoraPermission(
+        user=Some("Random_user_"+a),
+        roles=Some(randomRoles))
+      val randomTranslation=randomAgoraPerm.toFireCloudPermission
+      println("For Agora *RANDOM* "+randomAgoraPerm+" the FC translation is "+randomTranslation)
+
+    }
+    return false
+  }
+
+
+
+  def testFromAgoraToFC():Boolean = {
```

Comment:
This comment applies to all of the smaller tests in this method. Once the tests are pulled out and into a *Spec class, these should each be individual tests - if we really need them. I'm not really clear what it is you are trying to test. It's important to remember that we are unit testing Orchestration logic and not Agora.